plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/validation/SetMarkersAction.java

```diff
@@ -15,22 +15,23 @@
  ********************************************************************************/
 package org.eclipse.glsp.server.features.validation;
 
+import java.util.ArrayList;
 import java.util.List;
 
-import org.eclipse.glsp.server.actions.Action;
+import org.eclipse.glsp.server.actions.ResponseAction;
 
-public class SetMarkersAction extends Action {
+public class SetMarkersAction extends ResponseAction {
```

Comment:
Since you changed that into a response action I was worried that it might not properly work anymore to simply send that action from the server to the client without a request. However, as far as I can tell this still works since if there is no request for that the response id will be not set and the action is handled normally.

While I tested this I noticed that in `ResponseAction.respond` we do not filter for the respective reponse class, so if we have a request with id '1' and we return a list of responses they all get assigned the response id '1'. I think that may cause problems on the client where we wait for only one response for a request, so it may make sense to only assign the response id for one of the actions, what do you think? I believe something like this may work:

```
   public static Action respond(final Action request, final Action response) {
      if (request instanceof RequestAction<?>) {
         Class<?> responseClass = GenericsUtil.getGenericTypeParameterClass(request.getClass(), RequestAction.class);
         if (responseClass.isInstance(response)) {
            ((ResponseAction) response).setResponseId(((RequestAction<?>) request).getRequestId());
         }
      }
      return response;
   }
```

Reply:
Fixed